File: app/src/main/java/ru/starksoft/differ/sample/LoggerImpl.java
package ru.starksoft.differ.sample;

import android.util.Log;

import org.jetbrains.annotations.NotNull;

import androidx.annotation.NonNull;
import ru.starksoft.differ.Logger;

public final class LoggerImpl implements Logger {

	private static final Logger INSTANCE = new LoggerImpl();

	@NonNull
	public static Logger getInstance() {
		return INSTANCE;
	}

	@Override
	public void log(@NotNull String tag, @NotNull String message) {
		Log.d(tag, message);
	}

	@Override
	public void log(@NotNull String tag, @NotNull String message, @NotNull Throwable t) {
		Log.d(tag, message, t);
	}

	@Override
	public void w(@NotNull String tag, @NotNull String message) {
		Log.w(tag, message);
	}

	@Override
	public void d(@NotNull String tag, @NotNull String message) {
		Log.d(tag, message);
	}

	@Override
	public void e(@NotNull String tag, @NotNull String message, @NotNull Throwable t) {
		Log.e(tag, message, t);
	}
}

File: diff-adapter/src/main/java/ru/starksoft/differ/api/LoggerImpl.kt
package ru.starksoft.differ.api

import android.util.Log

class LoggerImpl : Logger {

	override fun w(tag: String, message: String) {
		Log.w(tag, message)
	}

	override fun d(tag: String, message: String) {
		Log.d(tag, message)
	}

	override fun e(tag: String, message: String, t: Throwable) {
		Log.e(tag, message, t)
	}

	companion object {
		@JvmStatic
		val instance: Logger = LoggerImpl()
	}
}
